elements/rh-chip/rh-chip.ts

```diff
@@ -0,0 +1,111 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { property } from 'lit/decorators/property.js';
+import { query } from 'lit/decorators/query.js';
+import { colorContextConsumer, type ColorTheme } from '../../lib/context/color/consumer.js';
+import { consume } from '@lit/context';
+
+import { context, type RhChipGroupContext } from './context.js';
+
+import '@rhds/elements/rh-icon/rh-icon.js';
+
+import styles from './rh-chip.css';
+
+export class ChipCheckedEvent extends CustomEvent<{ checked: boolean }> {
+  constructor(checked: boolean) {
+    super('chip-checked', {
+      bubbles: true,
+      composed: true,
+      detail: { checked },
+    });
+  }
+}
+
+/**
+ * Chip
+ * A chip is a clickable surface that acts like a checkbox.
+ * @summary Creates a checkable surface
+ * @fires {ChipCheckedEvent} chip-checked - when chip is checked/unchecked
+ * @csspart chip - The main chip container
+ * @slot - Place the label of the checkbox here
+ */
+@customElement('rh-chip')
+export class RhChip extends LitElement {
+  static readonly styles = [styles];
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  /**
+   * Whether the chip is checked.
+   */
+  @property({ type: Boolean, reflect: true }) checked = false;
+
+  /**
+   * Whether the chip is disabled.
+   */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /**
+   * Set a custom string for the input's `name` attribute. Defaults to `chip-checkbox`.
+   */
+  @property({ reflect: true, attribute: 'name' }) chipName?: string;
+
+  /**
+   * Set a custom string for the input's `value` attribute. Defaults to `on`.
+   */
+  @property({ reflect: true, attribute: 'value' }) chipValue?: string;
+
+  @query('input') private chipInput!: HTMLInputElement;
+
+  @consume({ context, subscribe: true })
+  @property({ attribute: false })
+  private ctx?: RhChipGroupContext;
+
+  render() {
+    const { on = 'light' } = this;
+    const attrName = this.chipName ? this.chipName : 'chip-checkbox';
+    const { size } = this.ctx ?? {};
+    return html`
+      <label part="chip" class=${classMap({ on: true, [on]: true, [`size-${size}`]: !!size })}>
+        <slot></slot>
+        <input type="checkbox"
+               name=${attrName}
+               value=${ifDefined(this.chipValue)}
+               @change=${this.#onChecked}
+               @keydown=${this.#handleKeydown}
+               ?checked=${!this.disabled && this.checked}
+               aria-disabled=${!!this.disabled}>
```

Comment:
```suggestion
               name="${attrName}"
               value="${ifDefined(this.chipValue)}"
               @change="${this.#onChecked}"
               @keydown="${this.#onKeydown}"
               ?checked="${this.checked}"
               aria-disabled="${String(this.disabled) as 'true'|'false'}">
```
why do we need name? it won't affect form submission

Reply:
[MDN states](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name):

> Consider the name a required attribute (even though it's not). If an input has no name specified, or name is empty, the input's value is not submitted with the form!

Curious to hear more of your reasoning behind getting rid of `name`.

Reply:
this input will not be submitted with any form, since it's in the shadow root. we'd need to set the name on the host and make it a FACE

Reply:
Good point. Based on previous specs from designers (ala [my comment below](https://github.com/RedHat-UX/red-hat-design-system/pull/2198#discussion_r1992281651)), this is not a FACE. I will remove the `name` attribute.

Reply:
Removed in [ea1dd8b](https://github.com/RedHat-UX/red-hat-design-system/pull/2198/commits/ea1dd8bf7ef5f63d4c3fa974c555021bfe7ffd55).